core/good_key.go

```diff
@@ -100,19 +172,29 @@ func GoodKeyRSA(key rsa.PublicKey) (err error) {
 	// The modulus SHOULD also have the following characteristics: an odd
 	// number, not the power of a prime, and have no factors smaller than 752.
 	// TODO: We don't yet check for "power of a prime."
+	if checkSmallPrimes(modulus) {
+		err = MalformedRequestError("Key divisible by small prime")
+		log.Debug(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func checkSmallPrimes(i *big.Int) bool {
```

Comment:
Document this function as to when it returns true or false. Or make it return an error, which makes it clear.